File: Data Structures & Algorithms/find-duplicate-integer/submission-5.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            int index=abs(nums[i])-1;
            if(nums[index]<0) return abs(nums[i]);
            nums[index]=-nums[index];
        }
        return -1;
    }
};

File: Data Structures & Algorithms/house-robber/submission-5.java
class Solution {
    int solve(int index,int[] nums,int[] dp){
        if(index==0) return nums[0];
        if(index<0) return 0;
        if(dp[index]!=-1) return dp[index];
        int take=nums[index]+solve(index-2,nums,dp);
        int notTake=solve(index-1,nums,dp);
        return dp[index]=Math.max(take,notTake);
    }
    public int rob(int[] nums) {
        int n=nums.length;
        int[] dp=new int[n];
        Arrays.fill(dp,-1);
        return solve(n-1,nums,dp);
    }
}
